### PR TITLE
Refactor Calligraphy into a ViewPump interceptor

### PR DESCRIPTION
**Pull request by jbarr21**
With the LayoutInflater and ContextWrapper moved to [ViewPump](https://github.com/InflationX/ViewPump/pull/2), Calligraphy can be refactored into a ViewPump interceptor.

This refactored version of Calligraphy will be namespaced as `calligraphy3` due to the major breaking changes.  This PR depends on SNAPSHOT version of ViewPump provided through `mavenLocal()`, but will update it when a hosted snapshot becomes available.

### CODE REVIEW

File: calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyConfig.java [hunk 1]
```diff
@@ -312,6 +224,11 @@ public Builder addCustomViewWithSetTypeface(Class<?> clazz) {
             return this;
         }
 
+        public Builder setFontMapper(FontMapper fontMapper) {
```
Comment (chrisjenx) [inline calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyConfig.java]: Can we move this FontMapper into a different PR?

File: calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyFactory.java [hunk 2]
```diff
@@ -15,6 +15,8 @@
 import java.lang.ref.WeakReference;
 import java.lang.reflect.Method;
 
+import io.github.inflationx.viewpump.ReflectionUtils;
+
 class CalligraphyFactory {
```
Comment (chrisjenx) [inline calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyFactory.java]: Do you think this should still be called `Factory`? The `ViewPump#Interceptor` calls into these methods right?

File: calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyFactory.java [hunk 3]
```diff
@@ -140,10 +144,12 @@ void onViewCreatedInternal(View view, final Context context, AttributeSet attrs)
                     textViewFont = CalligraphyUtils.pullFontPathFromTheme(context, styleForTextView[0], mAttributeId);
             }
 
+            textViewFont = applyFontMapper(textViewFont);
```
Comment (chrisjenx) [inline calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyFactory.java]: again pull out into a seperate PR

**Review by chrisjenx — CHANGES_REQUESTED** [review]
Let's also make this 14+ not that it probably makes much difference?

File: calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyConfig.java (hunk 1 above)
Comment (jbarr21) [reply calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyConfig.java]: sure, will do

File: calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyFactory.java (hunk 3 above)
Comment (jbarr21) [reply calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyFactory.java]: 👍

File: calligraphy/build.gradle [hunk 4]
```diff
@@ -21,6 +21,7 @@ android {
 }
 
 dependencies {
+    compile 'io.github.inflationx:viewpump:0.1.0-SNAPSHOT'
     compile 'com.android.support:appcompat-v7:24.0.0'
```
Comment (jbarr21) [inline calligraphy/build.gradle:25]: @chrisjenx what do you think about dependencies here? keep viewpump compile or change to provided?  what about appcompat, too?

File: calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyFactory.java (hunk 2 above)
Comment (jbarr21) [reply calligraphy/src/main/java/uk/co/chrisjenx/calligraphy3/CalligraphyFactory.java]: `Factory` doesn't make as much sense now since it's further from the `LayoutInflater.Factory`, but I can't think of a better name. maybe just `Calligraphy`?

**Comment by jbarr21** [comment]
> Let's also make this 14+ not that it probably makes much difference?

sounds good. I just kept it the same as the original lib, but happy to make the bump to 14

**Comment by jbarr21** [comment]
Addressed comments

**Review by chrisjenx — APPROVED** [review]
